travis/setup_dependencies_common.sh

```diff
@@ -120,6 +120,13 @@ if [[ $NUMPY_VERSION == dev* ]]; then
 elif [[ $NUMPY_VERSION == stable ]]; then
     conda install $QUIET numpy
     export CONDA_INSTALL="conda install $QUIET python=$PYTHON_VERSION"
+elif [[ $NUMPY_VERSION == pre* ]]; then
+    conda install $QUIET numpy
+    export CONDA_INSTALL="conda install $QUIET python=$PYTHON_VERSION"
+    if [[ -z $(pip list -o --pre | grep numpy | \
+            grep -E "[0-9]rc[0-9]|[0-9][ab][0-9]") ]]; then
+        exit
+    fi
```

Comment:
We want to stop the script if there isn't a pre-release available/installed (as in that case it would be just another build using the ``stable`` version). 

Here is a travis run when there was no pre-release on pypi:
https://travis-ci.org/astropy/ci-helpers/jobs/106326737

And here it runs:
https://travis-ci.org/astropy/ci-helpers/jobs/112297727